mirgecom/steppers.py

```diff
@@ -63,6 +63,14 @@ def _advance_state_stepper_func(rhs, timestepper, checkpoint, get_timestep,
         Time at which to start
     istep: int
         Step number from which to start
+    pre_step_callback
+        An optional user-defined function, with signature:
+        ``state = pre_step_callback(state, step, t, dt)``,
+        to be called before the timestepper is called for that particular step.
+    post_step_callback
+        An optional user-defined function, with signature:
+        ``state = post_step_callback(state, step, t, dt)``,
+        to be called after the timestepper is called for that particular step.
```

Comment:
Let's update the return signature here: ``state, dt = pre_step_callback(...)`` (same for post-step)